Make the tool gpg-like with its cli flags

ncdc opened the pull request:
Signed-off-by: Andy Goldstein <andy.goldstein@gmail.com>

ncdc committed:
Make the tool gpg-like with its cli flags

Signed-off-by: Andy Goldstein <andy.goldstein@gmail.com>

ncdc commented:
Looks like I need to add `--clearsign` support for deb-s3. Hang tight 😄

mattmoyer commented on main.go:
```diff
@@ -33,60 +39,263 @@ var (
-	PGP_UID_EMAIL (generate only):
-		Email for generated PGP key (e.g., "phil@example.com")
-`, os.Args[0])
+	pflag.Usage()
```
Maybe we could add a little extra usage blurb beyond what pflag returns here? I couldn't figure how to use `--export` correctly. Maybe add a message like `func usage(msg string) {` for clarity?

ncdc replied:
Sure

mattmoyer reviewed (commented):
Overall +1, will review again with the extra changes in just a sec.

ncdc committed:
Add clearsign support

Signed-off-by: Andy Goldstein <andy.goldstein@gmail.com>

ncdc committed:
Add usage help

Signed-off-by: Andy Goldstein <andy.goldstein@gmail.com>

ncdc commented:
@mattmoyer updated with clearsign support and slightly better usage output. One behavior that doesn't match `gpg` 100% - if you do `--clearsign --detach-sign`, because of the way my switch statement is coded, it will do a detached signature, but it won't clearsign it. The only way to clearsign is with non-detached signatures currently.

ncdc committed:
Update readme

Signed-off-by: Andy Goldstein <andy.goldstein@gmail.com>

ncdc committed:
Assume sign operation if only --detach-sign is passed

Signed-off-by: Andy Goldstein <andy.goldstein@gmail.com>

ncdc committed:
Various fixes

Support stdin for signing.

Support --default-key as a way to specify the signing key.

Default clearSign extension to .asc instead of .gpg.

Signed-off-by: Andy Goldstein <andy.goldstein@gmail.com>

mattmoyer reviewed (approved):
Thanks @ncdc!
